### PR TITLE
Add incrementalAlterConfigs to AdminClient

### PR DESCRIPTION
**Pull request by charlescd**

### CODE REVIEW

File: src/main/scala/zio/kafka/admin/AdminClient.scala [hunk 1]
```diff
@@ -804,6 +860,39 @@ object AdminClient extends Accessible[AdminClient] {
...
+    lazy val asJava: JAlterConfigOp = new JAlterConfigOp(configEntry, opType.asJava)
+  }
+
+  trait AlterConfigOpType {
+    def asJava: JAlterConfigOp.OpType
+  }
+
+  object AlterConfigOpType {
+    case object Set extends AlterConfigOpType {
+      lazy val asJava = JAlterConfigOp.OpType.SET
```
Comment (guizmaii) [inline src/main/scala/zio/kafka/admin/AdminClient.scala:880]: We don't need the lazy here

```suggestion
      val asJava = JAlterConfigOp.OpType.SET
```

File: src/main/scala/zio/kafka/admin/AdminClient.scala [hunk 2]
```diff
@@ -804,6 +860,39 @@ object AdminClient extends Accessible[AdminClient] {
...
+    }
+
+    case object Delete extends AlterConfigOpType {
+      lazy val asJava = JAlterConfigOp.OpType.DELETE
```
Comment (guizmaii) [inline src/main/scala/zio/kafka/admin/AdminClient.scala:884]:
```suggestion
      val asJava = JAlterConfigOp.OpType.DELETE
```

File: src/main/scala/zio/kafka/admin/AdminClient.scala [hunk 3]
```diff
@@ -804,6 +860,39 @@ object AdminClient extends Accessible[AdminClient] {
...
+    }
+
+    case object Append extends AlterConfigOpType {
+      lazy val asJava = JAlterConfigOp.OpType.APPEND
```
Comment (guizmaii) [inline src/main/scala/zio/kafka/admin/AdminClient.scala:888]:
```suggestion
      val asJava = JAlterConfigOp.OpType.APPEND
```

File: src/main/scala/zio/kafka/admin/AdminClient.scala [hunk 4]
```diff
@@ -804,6 +860,39 @@ object AdminClient extends Accessible[AdminClient] {
...
+    }
+
+    case object Substract extends AlterConfigOpType {
+      lazy val asJava = JAlterConfigOp.OpType.SUBTRACT
```
Comment (guizmaii) [inline src/main/scala/zio/kafka/admin/AdminClient.scala:892]:
```suggestion
      val asJava = JAlterConfigOp.OpType.SUBTRACT
```

File: src/main/scala/zio/kafka/admin/AdminClient.scala [hunk 5]
```diff
@@ -804,6 +860,39 @@ object AdminClient extends Accessible[AdminClient] {
...
+      val jOpts = new JAlterConfigsOptions().validateOnly(validateOnly)
+      timeout.fold(jOpts)(timeout => jOpts.timeoutMs(timeout.toMillis.toInt))
+    }
+  }
+
+  final case class AlterConfigOp(configEntry: ConfigEntry, opType: AlterConfigOpType) {
+    lazy val asJava: JAlterConfigOp = new JAlterConfigOp(configEntry, opType.asJava)
+  }
+
+  trait AlterConfigOpType {
```
Comment (guizmaii) [inline src/main/scala/zio/kafka/admin/AdminClient.scala]: 🌕 Should it be sealed?

**Review by guizmaii — COMMENTED** [review]
See comments

File: src/main/scala/zio/kafka/admin/AdminClient.scala (hunk 1 above)
Comment (charlescd) [reply src/main/scala/zio/kafka/admin/AdminClient.scala:880]: It is how `asJava` is implemented for every other case class/object. Even if I agree, I don't want to diverge from the original repo :)

File: src/main/scala/zio/kafka/admin/AdminClient.scala (hunk 2 above)
Comment (charlescd) [reply src/main/scala/zio/kafka/admin/AdminClient.scala:884]: See https://github.com/conduktor/zio-kafka/pull/98#discussion_r1006078579

File: src/main/scala/zio/kafka/admin/AdminClient.scala (hunk 3 above)
Comment (charlescd) [reply src/main/scala/zio/kafka/admin/AdminClient.scala:888]: See https://github.com/conduktor/zio-kafka/pull/98#discussion_r1006078579

File: src/main/scala/zio/kafka/admin/AdminClient.scala (hunk 4 above)
Comment (charlescd) [reply src/main/scala/zio/kafka/admin/AdminClient.scala:892]: See https://github.com/conduktor/zio-kafka/pull/98#discussion_r1006078579

**Review by mdulac — APPROVED** [review]

**Review by strokyl — APPROVED** [review]

**Pull request merged**